File: Codes/0724-find-pivot-index.py
'''
Author: LetMeFly
Date: 2024-07-08 13:17:16
LastEditors: LetMeFly
LastEditTime: 2024-07-08 13:18:26
'''
from typing import List

class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        sum_ =sum(nums)
        nowSum = 0
        for i in range(len(nums)):
            if sum_ - nums[i] == nowSum * 2:
                return i
            nowSum += nums[i]
        return -1

File: Codes/0724-find-pivot-index.java
/*
 * @Author: LetMeFly
 * @Date: 2024-07-08 13:18:49
 * @LastEditors: LetMeFly
 * @LastEditTime: 2024-07-08 13:26:52
 */
class Solution {
    public int pivotIndex(int[] nums) {
        int sum = 0;
        for (int i = 0; i < nums.length; i++) {
            sum += nums[i];
        }
        int nowSum = 0;
        for (int i = 0; i < nums.length; i++) {
            if (sum - nums[i] == nowSum * 2) {
                return i;
            }
            nowSum += nums[i];
        }
        return -1;
    }
}